generators/csharp/sdk/src/endpoint/AbstractEndpointGenerator.ts

```diff
@@ -39,7 +54,63 @@ export abstract class AbstractEndpointGenerator {
         };
     }
 
-    private getAllPathParameters({
+    protected getPagerEndpointSignatureInfo({
+        serviceId,
+        endpoint
+    }: {
+        serviceId: ServiceId;
+        endpoint: HttpEndpoint;
+    }): EndpointSignatureInfo {
+        const { pathParameters, pathParameterReferences } = this.getAllPathParameters({ serviceId, endpoint });
+        const request = getEndpointRequest({ context: this.context, endpoint, serviceId });
+        const requestParameter =
+            request != null
+                ? csharp.parameter({ type: request.getParameterType(), name: request.getParameterName() })
+                : undefined;
+        return {
+            baseParameters: [...pathParameters, requestParameter].filter((p): p is csharp.Parameter => p != null),
+            pathParameters,
+            pathParameterReferences,
+            request,
+            requestParameter,
+            returnType: this.getPagerReturnType(endpoint)
+        };
+    }
+
+    protected getPagerReturnType(endpoint: HttpEndpoint): csharp.Type {
+        const itemType = this.getPaginationItemType(endpoint);
+        const pager = this.context.getPagerClassReference({
+            itemType
+        });
+        return csharp.Type.reference(pager);
+    }
+
+    protected getPaginationItemType(endpoint: HttpEndpoint): csharp.Type {
+        this.assertHasPagination(endpoint);
+        const listItemType = this.context.csharpTypeMapper.convert({
+            reference: (() => {
+                switch (endpoint.pagination.type) {
+                    case "offset":
+                        return endpoint.pagination.results.property.valueType;
+                    case "cursor":
+                        return endpoint.pagination.results.property.valueType;
+                    default:
+                        assertNever(endpoint.pagination);
+                }
+            })(),
+            unboxOptionals: true
+        });
+
+        if (listItemType.internalType.type !== "list") {
+            throw new Error(
+                `Pagination result type for endpoint ${endpoint.name.originalName} must be a list, but is ${listItemType.internalType.type}.`
+            );
+        }
+        const itemType = listItemType.internalType.value;
+        return itemType;
```

Comment:
nit: We can return directly here:
```suggestion
        return listItemType.internalType.value;
```